servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultBlockingStreamingHttpRequest.java

```diff
@@ -201,8 +201,9 @@ public BlockingStreamingHttpRequest messageBody(final HttpMessageBodyIterable<Bu
         original.payloadHolder().messageBody(defer(() -> {
             HttpMessageBodyIterator<Buffer> body = messageBody.iterator();
             return fromIterable(() -> body)
-                    .map(o -> (Object) o)
-                    .concat(defer(() -> from(body.trailers()).filter(Objects::nonNull)));
+                    .cast(Object.class)
+                    .concat(defer(() -> from(body.trailers()).filter(Objects::nonNull).shareContextOnSubscribe()))
```

Comment:
For internal code, consider manual check for null and return `empty()` to reduce number of operators.
We can also consider adding `Publisher.fromNullable(...)` to simplify this use-case.

Reply:
I added a package private static `fromFilterNull` method for local usage. not sure yet we want to expose composite operators in the public API but we can always promote it in a followup PR.